matrix_sdk/src/http_client.rs

```diff
@@ -140,6 +143,35 @@ impl HttpClient {
             request
                 .try_into_http_request::<BytesMut>(&self.homeserver.to_string(), access_token)?
                 .map(|body| body.freeze())
+        } else {
+            // this should be unreachable since assert_identity on request_config can only be set
+            // with the appservice feature active
+            #[cfg(not(feature = "appservice"))]
+            return Err(HttpError::NeedsAppserviceFeature);
```

Comment:
Changed to `cfg`-based feature-switch, also allows restricting allow `assert_identity` things to the appservice feature. The catch here is that because of crate-interdependence currently the `appservice` feature seems to be always enabled when running commands in the crate root. So `cargo test` in the root will run with the `appservice` feature in `matrix-sdk` active. Something we could do is running `cargo test --workspace --exclude matrix-sdk-appservice` in CI and the appservice tests with an additional command? I'll open a PR to rework the `test-features` CI pipeline and try how that would look.

Reply:
I've integrated the CI changes here directly, with #226 being a possible cleanup to reduce CI time